### PR TITLE
WebGPURenderer: Fix NodeLibrary Logic for Minified Builds

### PR DESCRIPTION
**Pull request by RenaudRohlinger**
Related issue: #29187

**Description**

Issue:

Minifiers mangle class names like MeshStandardMaterial.name to shorter identifiers (e.g., a2), causing errors such as:

NodeMaterial: Material "MeshStandardMaterial" is not compatible.

Solution:

This PR fixes the issue by introducing a static type property on material classes. By using materialClass.type instead of materialClass.name, we prevent minification from breaking material compatibility checks.

PS:
Note: I strongly recommend merging this or a similar fix before releasing r169, as this issue currently breaks all Vite applications using the WebGPURenderer.

*This contribution is funded by [Utsubo](https://utsubo.com)*

### CODE REVIEW

File: src/materials/Material.js
Comment (RenaudRohlinger) [inline src/materials/Material.js:22]: Not a big fan of this but since we're not using es2020 we can't do:

class Material extends EventDispatcher {
    static type = 'MeshBasicMaterial';
    ...
}

and `MeshBasicMaterial.type = 'MeshBasicMaterial';` would break treeshaking I believe.

**Review by donmccurdy — APPROVED** [review]

**Comment by donmccurdy** [comment]
Let's merge this and get it into r170, if there are no objections — It would be really nice to support bundlers with WebGPU, blocked by this and #29156.

**Comment by Mugen87** [comment]
@RenaudRohlinger Just a heads up since I've almost overlooked the conversation in the commit: https://github.com/mrdoob/three.js/commit/ac963e25c6ddf9ea1f5f070e90e78529db96f701#r148635345

We might want to partly revert this PR by keeping the `type` member so it can be easier monkey-patched on app level.

**Pull request merged**
